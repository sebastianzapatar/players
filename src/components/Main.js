import React, { useEffect, useState } from 'react'

export const Main = () => {
    const [jugadores, setjugadores] = useState([]);
    const [respuesta, setrespuesta] = useState([]);
    const obtenerJugadores=async ()=>{
        const respuesta=await fetch('https://mach-eight.uc.r.appspot.com/');
        const api=await respuesta.json();
        const {values}=api;
        setjugadores(values); 
        
    }
    const heighSum=(heigh)=>{
        let datos=[];
        let clonejugadores=jugadores.sort((a,b)=>a.first_name.localeCompare(b.first_name));;
        for(let i=0;i<jugadores.length;i++){
            const altura=parseInt(jugadores[i].h_in);
            const busqueda=heigh-altura;
            const suma=clonejugadores.filter(elemento=>parseInt(elemento.h_in)===busqueda);
            for(let j=0;j<suma.length;j++){
                if(jugadores[i].first_name+" "+jugadores[i].last_name
                !==suma[j].first_name+" "+suma[j].last_name){
                    datos.push(jugadores[i].first_name+" "+jugadores[i].last_name +" "
                    +suma[j].first_name+" "+suma[j].last_name);
                }
                
            }
            clonejugadores= clonejugadores.filter(elemento=>elemento!==jugadores[i]);
        }
        return datos;
    }
    let renderizar="";
    if(respuesta.length===0)renderizar=<h1>{"No matches found"}</h1>;
    else{
        renderizar=respuesta.map((names) =>
        <li>{names}</li>)
    }
    const obtenerJugadoresAltura=(e)=>{
        const value=e.target.value;
        setrespuesta(heighSum(value));
    }
    
    useEffect(() => {
        obtenerJugadores();
    }, [])
    return (
        <div>
            <input type="number" 
            placeholder="Ingrese la suma de las alturas"
            onChange={obtenerJugadoresAltura}/><br/>
           
            {renderizar}
        </div>
    )
}
