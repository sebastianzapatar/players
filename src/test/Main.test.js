import {shallow} from 'enzyme';
import { Main } from "../components/Main";

describe('Test app',()=>{
    test('test1 render', () => {
        const wrapper=shallow(<Main/>);
        expect(wrapper).toMatchSnapshot();
    })
    test('test2 Not Matches found', () => {
        const wrapper=shallow(<Main/>);
        const text=wrapper.find('h1').text().trim();
        expect(text).toBe('No matches found');
    })
    test('Change in text', async() => {
        const wrapper=shallow(<Main/>);
        const value='0';
        wrapper.find('input').simulate('change',{target: {value}});
        expect(wrapper.find('h1').text()).toBe('No matches found');
    })
    test('Change2 in text', async() => {
        const wrapper=shallow(<Main/>);
        const value='-150';
        wrapper.find('input').simulate('change',{target: {value}});
        expect(wrapper.find('h1').text()).toBe('No matches found');
    })
    test('Change3 in text', async() => {
        const wrapper=shallow(<Main/>);
        const value='150';
        wrapper.find('input').simulate('change',{target: {value}});
        wrapper.contains('li')
    })
})