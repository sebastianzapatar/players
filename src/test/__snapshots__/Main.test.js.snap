// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app test1 render 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Ingrese la suma de las alturas"
    type="number"
  />
  <br />
  <button
    onClick={[Function]}
  >
    Simular 150
  </button>
  <h1>
    No matches found
  </h1>
</div>
`;
